packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -676,15 +677,21 @@ exports.extendNodeType = ({ type, store, reporter }) => {
 
     const { image, options } = args
     const {
-      file: { contentType },
+      file: { contentType, url: imgUrl, fileName: name },
     } = image
 
     if (contentType.indexOf(`image/`) !== 0) {
       return null
     }
 
-    const absolutePath = await cacheImage(store, image, options, reporter)
-    const extension = path.extname(absolutePath)
+    const url = createUrl(imgUrl, options)
+    const extension = path.extname(imgUrl)
```

Comment:
Does this work for all images coming from contentful?

Reply:
Will double check and tell you next week!

Reply:
Good catch. Actually that way was unreliable. I rewrote it to set the extension based on mime type. Now even a image with wrong file extension will be properly downloaded :)